### PR TITLE
add originalEvent to _clearMenus function

### LINKED ISSUE #26336: Add originalEvent to eventListeners
Add `originalEvent` to bootstrap `eventListeners` like hide.bs.dropdown, if support it.

I need to know, which element was clicked, before a dropdown was closed. I tried to add this code to the `Dropdown._clearMenus` function in `dropdown.js` file (after the definition of the `relatedTarget` variable).

`if (event) {
 relatedTarget.original = event.originalEvent;
}`

This code snippet make the `originalEvent` accessible from the `hide.bs.dropdown` event. For other events may be used similar way.

### PR DESCRIPTION
**Pull request by jakubhonisek**
fixes #26336

### CODE REVIEW

**Review by Johann-S — CHANGES_REQUESTED** [review]
Please fix this error: https://travis-ci.org/twbs/bootstrap/jobs/368542629#L1156

and you must add a unit test to cover this case

**Comment by jakubhonisek** [comment]
@Johann-S error was fixed

**Comment by Johann-S** [comment]
Please add a unit test

File: js/src/dropdown.js [hunk 1]
```diff
@@ -341,6 +341,10 @@ const Dropdown = (($) => {
           relatedTarget: toggles[i]
         }
 
+        if (event) {
+          relatedTarget.original = event.originalEvent
```
Comment (Johann-S) [inline js/src/dropdown.js]: `relatedTarget.originalEvent` would be better

**Comment by Johann-S** [comment]
Why did you limit this event to click ?

**Comment by jakubhonisek** [comment]
@Johann-S I wanted to avoid a collision with the name `originalEvent`. Also `click` is the only action that closes the dropdown (except `ESC` press).

File: js/src/dropdown.js [hunk 2]
```diff
@@ -341,6 +341,10 @@ const Dropdown = (($) => {
           relatedTarget: toggles[i]
         }
 
+        if (event && event.type === 'click') {
+          relatedTarget.mouseEvent = event
```
Comment (Johann-S) [inline js/src/dropdown.js]: I would prefer: `clickEvent`

File: js/tests/unit/dropdown.js [hunk 3]
```diff
@@ -498,6 +498,39 @@ $(function () {
     $dropdown.trigger('click')
   })
 
+  QUnit.test('should fire hide and hidden event with a mouseEvent', function (assert) {
```
Comment (Johann-S) [inline js/tests/unit/dropdown.js]: I you want to add this property only for click event, you should add a test which verify this property isn't added on other type of events

**Review by Johann-S — CHANGES_REQUESTED** [review]

File: js/tests/unit/dropdown.js [hunk 4]
```diff
@@ -498,6 +498,75 @@ $(function () {
...
+      .appendTo('#qunit-fixture')
+      .find('[data-toggle="dropdown"]')
+      .bootstrapDropdown()
+
+    $dropdown.parent('.dropdown')
+      .on('hide.bs.dropdown', function (e) {
+        assert.notOk(e.clickEvent)
+      })
+      .on('hidden.bs.dropdown', function (e) {
+        assert.onotO(e.clickEvent)
```
Comment (Johann-S) [inline js/tests/unit/dropdown.js]: `assert.onotO(e.clickEvent)` I don't think `onotO` exist on `assert` 😆

**Review by Johann-S — APPROVED** [review]
LGTM 👍 

@mdo can I add new features in our next release ? (4.1.1) Or should I wait for 4.2 ?

**Comment by XhmikosR** [comment]
@Johann-S: does this need to be documented in docs?

**Comment by Johann-S** [comment]
yep that can be usefull, good catch @XhmikosR 👍

**Comment by XhmikosR** [comment]
Closing for lack of replies.

**Comment by jakubhonisek** [comment]
Hi @XhmikosR , why did you close this pull request? Is there any action that I have to do? Or somebody else?

**Comment by XhmikosR** [comment]
https://github.com/twbs/bootstrap/pull/26343#issuecomment-399715984

**Pull request closed without merging**